libvips/foreign/heifsave.c

```diff
@@ -545,6 +545,12 @@ vips_foreign_save_heif_build( VipsObject *object )
 	heif->page_height = vips_image_get_page_height( heif->image );
 	heif->n_pages = heif->image->Ysize / heif->page_height;
 
+	if( (guint64) heif->page_width * heif->page_height > INT_MAX / 4 ||
+		heif->page_width > 16384 || heif->page_height > 16384 ) {
```

Comment:
This can be simplified to:
```suggestion
	if( heif->page_width > 16384 || heif->page_height > 16384 ) {
```
Given that 16384x16384 is lower than `INT_MAX / 4` anyway.